### PR TITLE
feat(ingestion-slowlane): Produce events with overriden keys to OVERFLOW topic

### LINKED ISSUE #14202: Send events that cross the capture endpoint threshold to slowlane

### PR DESCRIPTION
**Pull request by tomasfarias**
## Problem

From time to time, unique `(team_id, distinct_id)` pairs can experience spikes in traffic. These spikes can slow down our ingestion pipeline and have done so several times. To better deal with these spikes we randomly distribute these events across Kafka partitions:
* Via an automatic detection algorithm setup in #13934.
* Manually, using an environment variable to manually indicate `(team_id, distinct_id)` pairs to override.

However, both of these methods are not sufficient as they either require manual intervention, or cannot automatically detect all possible cases. For this reason, we are creating a new OVERFLOW topic to re-route these events. 

Closes https://github.com/PostHog/posthog/issues/14202

## Changes

This PR re-routes events that have been deemed to be randomly partitioned into this new topic.

## How did you test this code?

Unit test case.

### CODE REVIEW

**Comment by tomasfarias** [comment]
We'll have to merge this after the consumer side of things, as we don't want to start producing events that we cannot consume yet.

File: posthog/api/capture.py [hunk 1]
```diff
@@ -96,6 +96,8 @@ def log_event(data: Dict, event_name: str, partition_key: Optional[str]):
         KAFKA_SESSION_RECORDING_EVENTS
         if event_name in SESSION_RECORDING_EVENT_NAMES
         else settings.KAFKA_EVENTS_PLUGIN_INGESTION_TOPIC
+        if partition_key
+        else settings.KAFKA_EVENTS_PLUGIN_INGESTION_OVERFLOW_TOPIC
```
Comment (tiina303) [inline posthog/api/capture.py:100]: nit: I'd find it a little easier to read if we passed the kafka topic as a param to this function & then make 3 calls in `capture_internal`

**Review by tiina303 — APPROVED** [review]

File: posthog/api/capture.py (hunk 1 above)
Comment (tomasfarias) [reply posthog/api/capture.py]: I agree with you :+1:

File: posthog/api/capture.py [hunk 2]
```diff
@@ -485,15 +486,35 @@ def capture_internal(event, distinct_id, ip, site_url, now, sent_at, team_id, ev
...
+            event["event"],
+            kafka_topic=KAFKA_SESSION_RECORDING_EVENTS,
+            partition_key=kafka_partition_key,
+        )
+
+    if settings.INGESTION_OVERFLOW_ENABLED:
+        kafka_topic = settings.KAFKA_EVENTS_PLUGIN_INGESTION_OVERFLOW_TOPIC
+    else:
+        # If OVERFLOW is not enabled, we always produce to regular topic, as there will
+        # be no consumer to consume from it.
```
Comment (tiina303) [inline posthog/api/capture.py:503]: nit:
```suggestion
        # be no consumer to consume from the overflow topic.
```

File: posthog/api/test/test_capture.py [hunk 3]
```diff
@@ -1266,6 +1270,73 @@ def test_performance_events_go_to_session_recording_events_topic(self, kafka_pro
...
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK, response.content)
+
+        kafka_topic_used = kafka_produce.call_args_list[0][1]["topic"]
+        self.assertEqual(kafka_topic_used, KAFKA_EVENTS_PLUGIN_INGESTION_OVERFLOW)
+        key = kafka_produce.call_args_list[0][1]["key"]
+        self.assertEqual(key, None)
+
+    @patch("posthog.kafka_client.client._KafkaProducer.produce")
+    def test_overriden_keys_do_not_go_to_overflow_events_topic(self, kafka_produce):
```
Comment (tiina303) [inline posthog/api/test/test_capture.py:1308]:
```suggestion
    def test_events_go_to_normal_ingestion_topic_if_overflow_disabled(self, kafka_produce):
```

**Review by tiina303 — APPROVED** [review]

**Comment by tomasfarias** [comment]
We are deploying the PluginServer consumer first and then adding this on top. Not merging immediately after deploying as we want to test things out for a couple of days first.

**Pull request closed without merging**